src/arrayMethodSplice.js

```diff
@@ -4,8 +4,91 @@
  * Implement method Splice
  */
 function applyCustomSplice() {
+  [].__proto__.push2 = function(...forPushed) {
+    for (let i = 0; i < forPushed.length; i++) {
+      this[this.length] = forPushed[i];
+    }
+
+    return this;
+  };
+
   [].__proto__.splice2 = function(start, deleteCount, ...items) {
-    // write code here
+    const args = [...arguments];
+    const splisedEl = [];
+    const elAfterSpliced = [];
+
+    let startIndex = start;
+    let toDelete = deleteCount;
+
+    if (args.length === 0) {
+      return splisedEl;
+    }
+
+    if (args.length === 1) {
+      if (startIndex === undefined) {
+        startIndex = 0;
+      }
+
+      if (startIndex < 0) {
+        startIndex += this.length;
+      }
+
+      if (startIndex < 0) {
+        startIndex = 0;
+      }
+
+      toDelete = this.length;
+
+      if (startIndex > this.length) {
+        toDelete = 0;
+        startIndex = 0;
+      }
+    }
+
+    if (args.length > 1) {
+      if (startIndex === undefined && toDelete === undefined) {
+        return splisedEl;
+      }
+
+      if (startIndex === undefined) {
+        startIndex = 0;
+      }
+
+      if (startIndex < 0) {
+        startIndex += this.length;
+      }
+
```

Comment:
you repeat some checks several times, will be better to move them to separate function